third_party/move/move-model/src/model.rs

```diff
@@ -3500,12 +3510,19 @@ impl<'env> StructEnv<'env> {
 
     /// Gets a field by its offset.
     pub fn get_field_by_offset(&'env self, offset: usize) -> FieldEnv<'env> {
-        for data in self.data.field_data.values() {
-            if data.offset == offset {
-                return FieldEnv {
-                    struct_env: self.clone(),
-                    data,
-                };
+        self.get_field_by_offset_optional_variant(None, offset)
+    }
+
+    /// Gets a field by its offset, in context of an optional variant.
+    pub fn get_field_by_offset_optional_variant(
+        &'env self,
+        variant: Option<Symbol>,
+        offset: usize,
+    ) -> FieldEnv<'env> {
+        // TODO: speed this up via a cache RefCell<BTreeMap<(variant, offset), FieldId>>
```

Comment:
It is not worth an issue, so I removed the TODO. and just left as a comment that this can be sped up. It's a linear search over less then a dozen elements in average, so its OK for the model.